draft-ietf-tls-esni.md

```diff
@@ -1025,9 +996,31 @@ The backend server MUST NOT perform this operation if it negotiated TLS 1.2 or
 below. Note that doing so would overwrite the downgrade signal for TLS 1.3 (see
 {{RFC8446}}, Section 4.1.3).
 
-The "ech_is_inner" is expected to have an empty payload. If the payload is
-non-empty (i.e., the length of the "extension_data" field is non-zero) then the
-backend server MUST abort the handshake with an "illegal_parameter" alert.
+The payload of "encrypted_client_hello" is expected to be a `ClientECH` with
+`ClientECH.type == is-inner`. If this is not the case, the backend server MUST
+abort the handshake with an "illegal_parameter" alert.
+
+### Sending HelloRetryRequest {#backend-server-hrr}
+
+If the backend server sends HelloRetryRequest in response to the ClientHello, it
+confirms ECH acceptance as follows. It generates a message
+HelloRetryRequestECHConf equal to HelloRetryRequest, but with an
+"encrypted_client_hello" extension with as payload of 8 zero bytes. It then
+computes an 8-byte string
+
+~~~
+   hrr_accept_confirmation = HKDF-Expand-Label(0,
+      "hrr ech accept confirmation",
+      Transcript-Hash(ClientHelloInner...HelloRetryRequestECHConf),
+      8)
+~~~
+
+where HKDF-Expand-Label, Transcript.Hash, and "0" are as defined above and
+ClientHelloInner...HelloRetryRequestECHConf denotes the sequence of handshake
+messages that includes ClientHelloInner and HelloRetryRequestECHConf. Finally,
+the backend server constructs its HelloRetryRequest by overriding the payload of
```

Comment:
What does "overriding" mean here?

Reply:
I mean that it replaces the payload with the derived value. I changed "overriding" to "replacing".

Reply:
We might want to make it clear, in the definition of this extension, that it can exist in HelloRetryRequest, and in that case, carries `uint8 accept_signal[8]`. It just seems strange to say "we know the extension is defined with this structure, but in this exceptional case we just ignore that syntax and throw some bytes in the payload."

Reply:
I'm not understanding the recommendation. Do you mean we should say in {{encrypted-client-hello}} that the server's response may appear in HRR and define the structure?